# Cursul 2 #

## Variații ale modelului logistic ##

 În modelul logistic $P_{t+1}=P_t\left(1+r\left(1-\frac{P_t}{K}\right)\right)$. Adică scriind $P_{t+1}=F(P_t)$ avem $F(P)=\left(1+r\left(1-\frac{P}{K}\right)\right).$ Graficul acestei funcții este 
 
  ![](patr1.png) 
 
 Se observă din acest grafic  (sau dacă ați făcut experimente cu diferite valori ale lui $r$ sau $P_0$) că după un număr suficient de pași, $P_t$ poate deveni 
 negativ. O primă interpetare a acestui  fapt este că populația dispare. Dar mai realist este să presupunem că pentru valori din ce în ce mai mari ale lui $t$, $P_{t+1}$ scade dar rămîne pozitiv
 
 ![](exp.png)
 
 Aceasta este graficul funcției $F(P)=Pe^{r\left(1-\frac{P}{K}\right)}$. Modelul $P_{t+1}=P_te^{r\left(1-\frac{P}{K}\right)}$ se numește modelul logistic al lui Rickert, introdus prima dată îm 1954. Echilibrele acestuia sînt $P^*=0$ și $P^*=K$. Să motivăm introducerea acestui model. Pentru modelul logistic am luat
 $\frac{\Delta P}{P}=r\left(1-\frac{P}{K}\right)$ ca să obținem graficul 
 
 ![](lin.png)
 
 Una din problemele graficului este că nu putem avea că $\frac{\Delta P}{P}$ este mai mic decît $-1$. Aceasta ar însemna că avem mai mult de un deces per individ. Atunci este normal să presupunem că funcția $\frac{\Delta P}{P}$ are graficul de forma
 
 ![](exp-1.png)
 
 Acesta este graficul funcției de scădere exponențială, translatată cu $-1$. Deci obținem că $$\frac{\Delta P}{P}=ae^{-bP} - 1.$$ Facem substituțiile $b=\frac{r}{K}$ și $a=e^r$. Obținem deci $$\frac{\Delta P}{P}=e^r e^{-r\frac{P}{K}} - 1=e^{r\left(1-\frac{P}{K}\right)}-1.$$ Prin procedeul standard obținem că
 $$P_{t+1}=P_te^{r\left(1-\frac{P_t}{K}\right)}.$$
 
 Și în acest caz $K$ se interpretează drept capacitatea de încărcare a mediului, deoarece pentru $P > K$, $\Delta P < 0$. Rata de creștere intrinsecă este de data
 aceasta $e^r-1$.
 
 De fapt nici modelul Ricker nu este exact. Ca să obținem un model, măsurăm pentru fiecare $t$ valorile lui $P_t$ și $P_{t+1}$, desenăm punctele $(P_t,P_{t+1})$
pe un grafic (luăm $P_t$ pe axa $x-$lor, și $P_{t+1}$ pe axa $y-$lor) și „reglăm” parametrii $K$ și $r$ astfel ca graficul funcției să treacă cel mai aproape de puncte. 

Un alt model este $$P_{t+1}=\frac{\lambda P_t}{\left(1+aP_t\right)^\beta}.$$ Valorile parametrilor nu au o semnificație biologică clară, dar permit o ajustare mai 
bună.  


# Modele liniare multi-dimensionale #

Modele pe care le-am considerat pînă acum presupun că populația este formată dintr-un singur tip de indivizi. În realitate o populație conține subgrupuri a căror comportament vital este diferit. De exemplu insectele trec prin diferite faze de dezvoltare: ouă, larve, crisalide, adult. Ratele de deces variază pentru fiecare fază de dezvoltare și doar adulții se reproduc. La fel plantele trec prin stadiile de sămînță, sămînță încolțită, plantă neînflorită, plantă înflorită. 

Modelele liniare apar cînd nu studiem întreaga populație ci urmărim evoluția fiecărei subgrupe, precum grupele de vîrstă sau numărul de indivizi care se află într-o anumită etapă de dezvoltare. Vom folosi doar cele mai simple ecuații, ecuații liniare.

Să începem cu un exemplu. Avem o specie de insecte cu trei stadii de dezvoltare: ou, larvă și adult. Presupunem că după o unitate de timp un individ trece de la ou la larvă, de la larvă la adult după o altă unitate de timp. Adulții depun ouă și mor după o altă unitate de timp. Notăm astfel:
 * $E_t$ este numărul de ouă la momentul $t$;
 * $L_t$ este numărul de larve la momentul $t$;
 * $A_t$ este numărul de adulți la momentul $t$.
 Presupunem că doar $4\%$ din ouă devin larve, doar $39\%$ din larve devin adulți și fiecare adult produce în medie $73$ de ouă. Avem sistemul
 $$ \left\{ \begin{array}{ccc} E_t & = & 73 A_t\\ L_t & = & 0,04 E_t \\ A_t & = & 0,39 L_t. \end{array}\right.$$ Spunem că ecuațiile sînt **cuplate**. Ca o primă
 observație avem $A_{t+3}=0,39\cdot 0,04\cdot 73 A_t=1,1388 A_t$. 
 
 Dacă însă $65\%$ din adulți supraviețuiesc pentru încă o unitate de timp atunci sistemul devine $$ \left\{ \begin{array}{ccc} E_t & = & 73 A_t\\ L_t & = & 0,04 E_t \\ A_t & = & 0,39 L_t+0,65 A_t. \end{array}\right.$$ În acesată situație nu mai e clar cum putem reduce sistemul la o singură ecuație.
 
 Un alt exemplu este următorul: presupunem că într-o pădure sînt două specii de arbori. Notăm cu $A_t$, respectiv $B_t$ numărul de indivizi ai fiecărei specii într-un an. Cînd un arbore moare, un nou arbore crește în locul lui, dar nu neapărat din aceeași specie. Să presupunem că specia $A$ trăiește mai mult și are o rată de deces de $1\%$ pe an, iar $B$ are rata de deces de $5\%$. Pe de altă parte indivizii lui $B$ cresc mai repede și au șanse mai mari să ocupe un loc vacant: 
 $75\%$ din locuri vor fi ocupate de $B$ și $25\%$ de $A$. Sistemul care descrie acest fenomen este $$ \left\{\begin{array}{ccc} A_{t+1} & = & (0,99 +0,25 
 \cdot 0,01)A_t+0,25\cdot 0,05 B_t \\ B_{t+1} & = & 0,75\cdot 0,01 A_t + (0,95+0,75\cdot0,05) B_t. \end{array}\right.$$
 
 **Exercițiu:** Explicați de unde provin coeficienții sistemului.
 
 După ce efectuăm calculele obținem $$ \left\{\begin{array}{ccc} A_{t+1} & = & 0,9925 A_t+0,0125 B_t \\ B_{t+1} & = & 0,0075 A_t + 0,9875 B_t. \end{array}\right.$$
 Nu e deloc clar cum am putea rezolva acest sistem cu metode elementare. Putem însă să calculăm un număr de pași pentru diferite valori inițiale.

In [30]:
using PrettyTables,IJulia 
A = IJulia.readprompt("A0= ")
A = parse(Float64, A)
B = IJulia.readprompt("B0= ")
B = parse(Float64, B)
tstop = IJulia.readprompt("Timpul final = ")
tstop = parse(Int64,tstop)
vals=[(A,B)]
for t in 1:tstop
    A1=0.9925*A+0.0125*B
    B1=0.0075*A+0.9875*B
    append!(vals,[(A1,B1)])
    A=A1
    B=B1
end
df=hcat(collect(0:tstop),[vals[i][1] for i in 1:length(vals)],[vals[i][2] for i in 1:length(vals)])
conf=set_pt_conf(tf = tf_markdown, alignment = :c);
header=["Timp", "A","B"];
pretty_table_with_conf(conf,df;header=header)

A0=  10
B0=  990
Timpul final =  1000


|  Timp  |    A    |    B    |
|--------|---------|---------|
|  0.0   |  10.0   |  990.0  |
|  1.0   |  22.3   |  977.7  |
|  2.0   | 34.354  | 965.646 |
|  3.0   | 46.1669 | 953.833 |
|  4.0   | 57.7436 | 942.256 |
|  5.0   | 69.0887 | 930.911 |
|  6.0   | 80.2069 | 919.793 |
|  7.0   | 91.1028 | 908.897 |
|  8.0   | 101.781 | 898.219 |
|  9.0   | 112.245 | 887.755 |
|  10.0  |  122.5  |  877.5  |
|  11.0  | 132.55  | 867.45  |
|  12.0  | 142.399 | 857.601 |
|  13.0  | 152.051 | 847.949 |
|  14.0  | 161.51  | 838.49  |
|  15.0  | 170.78  | 829.22  |
|  16.0  | 179.864 | 820.136 |
|  17.0  | 188.767 | 811.233 |
|  18.0  | 197.492 | 802.508 |
|  19.0  | 206.042 | 793.958 |
|  20.0  | 214.421 | 785.579 |
|  21.0  | 222.633 | 777.367 |
|  22.0  | 230.68  | 769.32  |
|  23.0  | 238.566 | 761.434 |
|  24.0  | 246.295 | 753.705 |
|  25.0  | 253.869 | 746.131 |
|  26.0  | 261.292 | 738.708 |
|  27.0  | 268.566 | 731.434 |
|  28.0  | 275.695 | 724.305 |
|  29.0  | 282.681 | 717.319 |
|  30.0 

Din tabelul de mai sus pare că sistemul evoluează spre un echilibru $(625,375)$. Dacă luăm alte valori ințiale vom obține un alt echilbru ale cărui componente satisfac $\frac{5}{3}$. Ca să înțelegem acest fenomen avem nevoie de noțiuni și metode noi.

## Matrice și vectori ##

**Definiție**: Un vector din $\mathbb{R}^n$ este o listă de $n$ numere reale, de obicei puse pe verticală.

Spre exemplu $\left(\begin{array}{c} 1 \\ 2 \\-8 \end{array}\right)$ este un vector din $\mathbb{R}^3$, $\left(\begin{array}{c} 1,2 \\ 2 \\-8 \\ 3,1 \end{array}\right)$ este un vector din $\mathbb{R}^4$. etc. De obicei notăm un vector cu litere bold $\mathbf{x}$. Pentru sistemul de mai sus putem lua $\mathbf{x}_t= 
\left(\begin{array}{c} A_t \\ B_t \end{array}\right)$, astfel că $$\mathbf{x}_{600}= 
\left(\begin{array}{c} 624,998 \\ 375,003 \end{array}\right).$$

**Definiție:** O matrice de tip $m \times n$ cu este un tablou dreptunghiular de numere reale cu $m$ linii și $n$ coloane.

Exemple de matrice: $\left(\begin{array}{cc} 1 & 2 \\ 2 & 1 \\-8  & 0\end{array}\right)$ este o matrice de tip $3\times 2$; $\left(\begin{array}{cc} 1 & 2 & 2 \\ 2 & 1 & 0\\-8  & 0 & 1\end{array}\right)$ este o matrice de tip $3 \times 3$. Ultima este o matrice *pătratică*. Se observă că un vector este o matrice de tip 
$n \times 1$. Matricele se notează cu litere mari. Pentru sistemul nostru $$P=\left(\begin{array}{cc} 0,9925 & 0,0125 \\ 0,0075 & 0,9875 \end{array}\right)$$ și se numește matricea de proiecție sau de tranziție a sistemului. Folosind vectori și matrici putem scrie sistemul $$\mathbf{x}_{t+1}=P \cdot \left(\begin{array}{c} A_t \\ B_t \end{array}\right)$$ sau chiar mai scurt $x_{t+1} = P \cdot x_t.$

**Definiție:** Dacă avem o matrice $A=\left(\begin{array}{cc} a & b \\ c & d \end{array}\right)$ și un vector $\mathbf{x}=\left(\begin{array}{cc} e \\ f \end{array}\right).$ Atunci $A\cdot\mathbf{x}=\left(\begin{array}{cc} ae + bf \\ cf+de \end{array}\right).$ 

Să modificăm modelul nostru: presupunem că matricea de tranziție corespunde unui an ploios, și că într-un an secetos avem matricea de tranziție $$S=\left(\begin{array}{cc} 0,9925 & 0,0975 \\ 0,0075 & 0,9025 \end{array}\right).$$ Aici am modificat mortalitatea lui $B$ la $0,39.$ Presupunem că anii ploioși alternează cu cei secetoși. Fie $$\mathbf{x}_0=\left(\begin{array}{cc} 10 \\ 990 \end{array}\right)$$ și în plus primul an este secetos. Atunci $$\mathbf{x}_1=S\cdot\mathbf{x_0}=\left(\begin{array}{cc} 0,9925\cdot 10 + 0,0975\cdot 990 \\ 0,0075\cdot 10+0,9025\cdot 990 \end{array}\right)=\left(\begin{array}{cc} 106,45 \\ 893,55 \end{array}\right).$$ Presupunem că anul următor este ploios atunci $$\mathbf{x}_2=P\cdot\mathbf{x}_1=\left(\begin{array}{cc} 0,9925 & 0,0125 \\ 0,0075 & 0,9875 \end{array}\right)\cdot\left(\begin{array}{c} 106,45 \\ 893,55 \end{array}\right)=P\cdot(S\cdot \mathbf{x}_0).$$ Se pune întrebarea dacă există p matrice
$B$ cu proprietatea că $\mathbf{x}_2=B\cdot\mathbf{x}_0$. 

O astfel de matrice există și este produsul celor două matrici: $B=P\cdot S$.

**Definiție:** Fie $A=\left(\begin{array}{cc} a & b \\ c & d \end{array}\right)$ și $A=\left(\begin{array}{cc} e & f \\ g & h \end{array}\right)$. Atunci
$$A\cdot B = A=\left(\begin{array}{cc} ae+bg & af+bh \\ ec+dg & cf+dh \end{array}\right).$$ 